### ISSUE TITLE
@cds/core: `cds-control` can cause overlay elements to appear underneath it

### ISSUE THREAD

**Issue by astorije** [vmware/clarity#6220]
## Describe the bug

TL;DR: This is the exact same issue as https://github.com/vmware/clarity/issues/5710 / https://github.com/vmware/clarity/pull/5852

When a custom control contains an overlay element directly located in its DOM, the overlay always appears underneath other `cds-control`s on the page below it.

## How to reproduce

https://codesandbox.io/s/cds-control-contain-bug-hi4fp

Open the "Click me" dropdown and observe the layout issue:

<img width="425" alt="Screen Shot 2021-08-05 at 12 30 57 PM" src="https://user-images.githubusercontent.com/113730/128387995-35ce7070-310f-4eff-9820-d540aa505f27.png">

## Expected behavior

The overlay dropdown should appear as such:

<img width="462" alt="Screen Shot 2021-08-05 at 12 30 40 PM" src="https://user-images.githubusercontent.com/113730/128388081-69495658-1ec0-4ed0-a5f1-5ba0e7933862.png">

This screenshot was generated by overriding the CSS as such:

```css
cds-control {
  contain: inherit;
}
```

## Versions

**Clarity project:**

- [x] Clarity Core
- [ ] Clarity Angular/UI

**Clarity version:**

- [ ] v3.x
- [ ] v4.x
- [x] v5.5.0

**Framework:**

- [ ] Angular
- [ ] React
- [ ] Vue
- [x] Other: All

## Additional notes

I am more than happy to open a PR similar to https://github.com/vmware/clarity/pull/5852/files on [`control.element.scss#L9`](https://github.com/vmware/clarity/blob/next/packages/core/src/forms/control/control.element.scss#L9) if you think this is an acceptable solution.

**Comment by mathisscott** [comment]
The dropdowns we have in progress in Clarity Core don't have this problem. So that's a light at the end of the tunnel, so to speak.

As for adding `contain: inherit`, my only qualm is whether that works in Safari. Might be worth a look before submitting a PR.

Otherwise, I think this is an acceptable solution and thanks in advance.

CCing @coryrylan for his input too.

**Comment by coryrylan** [comment]
I think this fix makes sense for the generic control container. Since all controls inherit from this control they will also inherit the `contain: inherit` style. This should be fine for the other controls. We might lose a small performance gain using it for the other controls but we can adjust it per form control use case or once the core dropdown is complete.

**Comment by astorije** [comment]
Good call re: Safari! Thankfully the bug doesn't appear there, so both with and without the extra CSS declaration, the result is as follows:

<img width="520" alt="Screen Shot 2021-08-06 at 4 17 06 PM" src="https://user-images.githubusercontent.com/113730/128566709-217987db-2c31-49c0-912b-aade8f0c89d2.png">

**Comment by astorije** [comment]
Thanks for the feedback, I opened a PR at https://github.com/vmware/clarity/pull/6237
